test/unit/ConstantProductPair.t.sol

```diff
@@ -305,8 +305,7 @@ contract ConstantProductPairTest is BaseTest, IReservoirCallee {
 
         // act
         for (uint256 i = 0; i < lMaxObservations + 4; ++i) {
-            vm.roll(block.number + 1);
-            vm.warp(block.timestamp + 5);
+            _stepTime(5);
```

Comment:
this is a weird bug that popped up after switching to `via-ir`. To me the two implementations are 100% the same, but somehow the previous one gave an assertion failure in the only assertion of the test, when built with `via-ir`

Reply:
Nice that you isolated to a single commit (16d043e201853d0724d3d9c4fdc23bcab57c9ee5), this could be a bug in `solc`, there have been quite a few optimizer bugs... which means we should only use via-ir if gas saving is fairly significant

Reply:
We should consider reporting this bug to `solc`. We could also diff the assembly produced

Reply:
I've tried `forge debug` that specific test but somehow the debugger just doesn't even run, so I'm trying to get the compiler to do output some asm now

Reply:
should be an `--extra-output-file evm.asm` option you can use. then diff the output

Reply:
I generated both assemblies and tried to diff them [here](https://github.com/vexchange/v3-core/tree/bug/via-ir/asm). Not sure if this way of diffing is the most useful, and what's the next step from here. 

In the diff file in that branch I've deleted a lot of things that are obviously not useful

Reply:
lmao, that diff resulted in 8000 lines different? if you want i can look into this for you

Reply:
Given that this bug doesn't affect our core code's correctness, it's not super urgent to me right now.

Honestly I want to contribute back to the solidity community and report this bug, but I don't know how to make sense of the diff of the assembly output for now. 

So I leave it up to you whether you want to investigate further and get the real diff of the opcodes. I don't mind filling the bug report once we have distilled the essence of the diff

Reply:
Agreed